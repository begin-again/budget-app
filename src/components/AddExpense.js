import React from 'react';

const AddExpense = () => {
  const template =
    <div>
        Add Expense component
    </div>
        ;
  return template;
};

export default AddExpense;
