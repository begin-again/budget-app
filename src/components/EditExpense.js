import React from 'react';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const EditExpense = (props) => {
  const template =
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          console.log('updated', expense);
          props.dispatch(editExpense(props.expense.id, expense));
          props.history.push('/');
        }}
      />
      <button onClick={(e) => {
        props.dispatch(removeExpense({ id: props.expense.id }));
        props.history.push('/');
      }}
      >Remove
      </button>
    </div>
        ;
  return template;
};

const mapStateToProps = (state, props) => (
  {
    expense: state.expenses.find(expense => expense.id === props.match.params.id)
  }
);

export default connect(mapStateToProps)(EditExpense);
