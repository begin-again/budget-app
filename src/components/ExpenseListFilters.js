import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount } from '../actions/filters';

const ExpenseListFilters = (props) => (
  <div>
    <input
      type='text' placeholder='filter by'
      value={props.filters.text}
      onChange={(event) => {
        props.dispatch(setTextFilter(event.target.value));
      }}
    />
    <select
      value={props.filters.sortBy}
      onChange={(e) => {
        if (e.target.value === 'date') {
          props.dispatch(sortByDate());
        } else {
          props.dispatch(sortByAmount());
        }
      }}
    >
      <option value='amount'>Amount</option>
      <option value='date'>Date</option>
    </select>
  </div>
);

const mapStateToProps = (state) => (
  {
    filters: state.filters
  }
);

export default connect(mapStateToProps)(ExpenseListFilters);
