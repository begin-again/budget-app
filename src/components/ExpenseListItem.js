import React from 'react';
import { Link } from 'react-router-dom';

export default ({ description, amount, createdAt, id }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>
        {description}
      </h3>
    </Link>
    <p>{amount} &raquo; {createdAt}</p>
  </div>
);
