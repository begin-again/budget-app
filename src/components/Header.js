import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => {
  const template =
    <header>
      <h1>Expensify</h1>
      <NavLink to='/dashboard' activeClassName='is-active' exact>Dashboard</NavLink>
      <NavLink to='/create' activeClassName='is-active'>Create Expense</NavLink>
    </header>
        ;
  return template;
};

export default Header;
