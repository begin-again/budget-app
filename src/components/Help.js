import React from 'react';

const Help = () => {
  const template =
    <div>
        Help component
    </div>
        ;
  return template;
};

export default Help;
