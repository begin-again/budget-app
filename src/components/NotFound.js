import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => {
  const template =
    <div>
        404! - <Link to='/'>Go Home</Link>
    </div>
        ;
  return template;
};

export default NotFound;
