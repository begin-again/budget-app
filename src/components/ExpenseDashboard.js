import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboard = () => {
  const template =
    <div>
      <ExpenseListFilters />
      <ExpenseList />
    </div>
        ;
  return template;
};

export default ExpenseDashboard;
