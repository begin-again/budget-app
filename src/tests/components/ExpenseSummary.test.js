/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

describe('Expenses Summary Component', () => {
  it('should render with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={20000} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={22} expensesTotal={12345600} />);

    expect(wrapper).toMatchSnapshot();
  });
});
