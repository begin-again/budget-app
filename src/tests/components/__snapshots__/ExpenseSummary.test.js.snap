// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses Summary Component should render with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $200.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Expenses Summary Component should render with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        22
      </span>
       
      expenses
       totalling 
      <span>
        $123,456.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
